File: mRemoteV1/Forms/frmChoosePanel.vb
Imports mRemote3G.App

Namespace Forms
    Public Class frmChoosePanel
        Public Property Panel As String
            Get
                Return cbPanels.SelectedItem.ToString
            End Get
            Set
                cbPanels.SelectedItem = value
            End Set
        End Property

        Private Sub frmChoosePanel_Load(sender As Object, e As EventArgs) Handles MyBase.Load
            ApplyLanguage()

            AddAvailablePanels()
        End Sub

        Private Sub ApplyLanguage()
            btnOK.Text = Language.Language.strButtonOK
            lblDescription.Text = Language.Language.strLabelSelectPanel
            btnNew.Text = Language.Language.strButtonNew
            btnCancel.Text = Language.Language.strButtonCancel
            Text = Language.Language.strTitleSelectPanel
        End Sub

        Private Sub AddAvailablePanels()
            cbPanels.Items.Clear()

            For i = 0 To Runtime.WindowList.Count - 1
                cbPanels.Items.Add(Runtime.WindowList(i).Text.Replace("&&", "&"))
            Next

            If cbPanels.Items.Count > 0 Then
                cbPanels.SelectedItem = cbPanels.Items(0)
                cbPanels.Enabled = True
                btnOK.Enabled = True
            Else
                cbPanels.Enabled = False
                btnOK.Enabled = False
            End If
        End Sub

        Private Sub btnNew_Click(sender As Object, e As EventArgs) Handles btnNew.Click
            Dim pnlName As String = InputBox(Language.Language.strPanelName & ":", Language.Language.strNewPanel,
                                             Language.Language.strNewPanel)

            If pnlName <> "" Then
                Runtime.AddPanel(pnlName)
                AddAvailablePanels()
                cbPanels.SelectedItem = pnlName
                cbPanels.Focus()
            End If
        End Sub

        Private Sub btnOK_Click(sender As Object, e As EventArgs) Handles btnOK.Click
            Me.DialogResult = DialogResult.OK
        End Sub

        Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
            Me.DialogResult = DialogResult.Cancel
        End Sub
    End Class
End NameSpace

File: mRemoteV1/CS/Forms/frmChoosePanel.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

//using mRemoteNG.App.Runtime;

namespace mRemoteNG
{
	public partial class frmChoosePanel
	{
		public frmChoosePanel()
		{
			InitializeComponent();
		}
public string Panel
		{
			get
			{
				return cbPanels.SelectedItem.ToString();
			}
			set
			{
				cbPanels.SelectedItem = value;
			}
		}
		
		public void frmChoosePanel_Load(System.Object sender, System.EventArgs e)
		{
			ApplyLanguage();
			
			AddAvailablePanels();
		}
		
		private void ApplyLanguage()
		{
			btnOK.Text = My.Language.strButtonOK;
			lblDescription.Text = My.Language.strLabelSelectPanel;
			btnNew.Text = My.Language.strButtonNew;
			btnCancel.Text = My.Language.strButtonCancel;
			Text = My.Language.strTitleSelectPanel;
		}
		
		private void AddAvailablePanels()
		{
			cbPanels.Items.Clear();
			
			for (int i = 0; i <= WindowList.Count - 1; i++)
			{
				cbPanels.Items.Add(WindowList[i].Text.Replace("&&", "&"));
			}
			
			if (cbPanels.Items.Count > 0)
			{
				cbPanels.SelectedItem = cbPanels.Items[0];
				cbPanels.Enabled = true;
				btnOK.Enabled = true;
			}
			else
			{
				cbPanels.Enabled = false;
				btnOK.Enabled = false;
			}
		}
		
		public void btnNew_Click(System.Object sender, System.EventArgs e)
		{
			string pnlName = Interaction.InputBox(My.Language.strPanelName + ":", My.Language.strNewPanel, My.Language.strNewPanel);
			
			if (!string.IsNullOrEmpty(pnlName))
			{
				AddPanel(pnlName);
				AddAvailablePanels();
				cbPanels.SelectedItem = pnlName;
				cbPanels.Focus();
			}
		}
		
		public void btnOK_Click(System.Object sender, System.EventArgs e)
		{
			this.DialogResult = System.Windows.Forms.DialogResult.OK;
		}
		
		public void btnCancel_Click(System.Object sender, System.EventArgs e)
		{
			this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
		}
	}
}
